docs/core/extensions/queue-service.md

```diff
@@ -0,0 +1,116 @@
+You may be familiar with the <xref:System.Web.Hosting.HostingEnvironment.QueueBackgroundWorkItem(System.Func{System.Threading.CancellationToken,System.Threading.Tasks.Task})> functionality from the `System.Web.Hosting` namespace. To model a service that is inspired by this functionality, start by adding an `IBackgroundTaskQueue` interface to the project:
+
+:::code source="snippets/workers/queue-service/IBackgroundTaskQueue.cs":::
+
+There are two methods, one which exposes queuing functionality and the other that dequeues previously queued work items. A *work item* is a `Func<CancellationToken, ValueTask>`. Next, add the default implementation to the project.
+
+:::code source="snippets/workers/queue-service/DefaultBackgroundTaskQueue.cs":::
+
+The preceding implementation relies on a <xref:System.Threading.Channels.Channel%601> as a queue. The <xref:System.Threading.Channels.BoundedChannelOptions.%23ctor(System.Int32)> is called with an explicit capacity. Capacity should be set based on the expected application load and number of concurrent threads accessing the queue. <xref:System.Threading.Channels.BoundedChannelFullMode.Wait?displayProperty=nameWithType> will cause calls to <xref:System.Threading.Channels.ChannelWriter%601.WriteAsync%2A?displayProperty=nameWithType> to return a task, which completes only when space became available. This leads to backpressure, in case too many publishers/calls start accumulating.
+
+## Rewrite the Worker class
+
+In the following `QueueHostedService` example:
+
+- The `ProcessTaskQueueAsync` method returns a <xref:System.Threading.Tasks.Task> in `ExecuteAsync`.
+- Background tasks in the queue are dequeued and executed in `ProcessTaskQueueAsync`.
+- Work items are awaited before the service stops in `StopAsync`.
+
+Replace the existing `Worker` class with the following C# code, and rename the file to *QueueHostedService.cs*.
+
+:::code source="snippets/workers/queue-service/QueuedHostedService.cs" highlight="35-36,38":::
+
+A `MonitorLoop` service handles enqueuing tasks for the hosted service whenever the `w` key is selected on an input device:
+
+- The `IBackgroundTaskQueue` is injected into the `MonitorLoop` service.
+- `IBackgroundTaskQueue.QueueBackgroundWorkItemAsync` is called to enqueue a work item.
+- The work item simulates a long-running background task:
+  - Three 5-second delays are executed <xref:System.Threading.Tasks.Task.Delay%2A>.
+  - A `try-catch` statement traps <xref:System.OperationCanceledException> if the task is cancelled.
+
+:::code source="snippets/workers/queue-service/MonitorLoop.cs" highlight="11,16,41":::
+
+Replace the existing `Program` contents with the following C# code:
+
+:::code source="snippets/workers/queue-service/Program.cs" highlight="8-18":::
+
+The services are registered in `IHostBuilder.ConfigureServices` (*Program.cs*). The hosted service is registered with the `AddHostedService` extension method. `MonitorLoop` is started in *Program.cs* top-level statement:
```

Comment:
Does this article assume readers are familiar with DI conventions? If not, a link to this article would be helpful if they want to learn more. 

https://docs.microsoft.com/dotnet/core/extensions/dependency-injection

Reply:
Yes, it's near the same TOC node - logically following it. And is explicitly communicated in the overview / introductory article.